### PR TITLE
kubernetes: fix printf format errors

### PR DESCRIPTION
**Pull request by rsc**
These are all flagged by Go 1.11's
more accurate printf checking in go vet,
which runs as part of go test.

```release-note
NONE
```

### CODE REVIEW

**Comment by rsc** [comment]
I think I completed the CLA dance, although note that I used rsc@golang.org (my usual work-on-Go email address) for the commit instead of rsc@google.com. I would prefer to keep using rsc@golang.org but I can flip the commit to rsc@google.com if needed. My understanding is that posting this comment should rerun the CLA check somehow?

**Comment by rsc** [comment]
/assign @brendandburns @deads2k @jsafrane 
as suggested by k8s-ci-robot

**Comment by rsc** [comment]
Ran gofmt with Go 1.10.3 and re-pushed. Note that I think those files were formatted with Go 1.11 beta 1 or so, so that means you'll see a tiny bit of gofmt drift as you move to Go 1.11.

**Comment by rsc** [comment]
/test pull-kubernetes-kubemark-e2e-gce-big

(I can't see how changing these buggy logging prints would cause a timeout, so I am guessing this is a flake.)

**Comment by rsc** [comment]
Whoops, GitHub did not show me @idealhack's rerun from 9 hours ago. Sigh.

**Comment by idealhack** [comment]
@rsc We can wait the OWNERS’ approve if these tests continue being flaky. There will be a bot which helps rerunning them.

**Comment by rsc** [comment]
/assign @brendandburns
as suggested by k8s-ci-robot

File: pkg/cloudprovider/providers/azure/azure_vmss_cache.go
Comment (andyzhangx) [inline pkg/cloudprovider/providers/azure/azure_vmss_cache.go]: it should be `v.Name`, pls revert this code change

**Review by andyzhangx — CHANGES_REQUESTED** [review]
reviewed azure related code

File: pkg/cloudprovider/providers/azure/azure_vmss_cache.go
Comment (rsc) [reply pkg/cloudprovider/providers/azure/azure_vmss_cache.go]: `vm.Name` is a `*string` (pointer to string), not a string, and `%q` will not print it. I thought about using `*vm.Name` but I didn't want to turn what used to be a buggy warning into an actual crash (if `vm.Name` was nil). 

Should I use `*vm.Name`? Is that guaranteed to be non-nil?

Comment (andyzhangx) [reply pkg/cloudprovider/providers/azure/azure_vmss_cache.go]: @rsc Then I would suggest keep the original code:
```
glog.Warningf("failed to get computerName for vmssVM (%q)", vm.Name)
```

This is `fix printf format errors` PR, not bug fix.

Comment (rsc) [reply pkg/cloudprovider/providers/azure/azure_vmss_cache.go]: @andyzhangx Yes, this is a "fix printf format errors" PR. 

I am editing that line ***because it has a printf format error***. 

Compiled using Go 1.11, the pkg/cloudprovider/providers/azure test fails (because test implies vet for printf format errors, and go vet has gotten better at detecting printf wrappers like glog.Warningf):

```
# k8s.io/kubernetes/pkg/cloudprovider/providers/azure
pkg/cloudprovider/providers/azure/azure_client.go:848: Verbose.Infof format %q reads arg #2, but call has 1 arg
pkg/cloudprovider/providers/azure/azure_client.go:850: Verbose.Infof format %q reads arg #2, but call has 1 arg
pkg/cloudprovider/providers/azure/azure_client.go:879: Verbose.Infof format %q has arg VMInstanceIDs of wrong type k8s.io/kubernetes/vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2017-12-01/compute.VirtualMachineScaleSetVMInstanceRequiredIDs
pkg/cloudprovider/providers/azure/azure_client.go:881: Verbose.Infof format %q has arg VMInstanceIDs of wrong type k8s.io/kubernetes/vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2017-12-01/compute.VirtualMachineScaleSetVMInstanceRequiredIDs
pkg/cloudprovider/providers/azure/azure_loadbalancer.go:265: Verbose.Infof format %s has arg isInternal of wrong type bool
pkg/cloudprovider/providers/azure/azure_loadbalancer.go:845: Verbose.Infof format %q reads arg #3, but call has 2 args
pkg/cloudprovider/providers/azure/azure_vmss.go:478: Verbose.Info call has possible formatting directive %q
pkg/cloudprovider/providers/azure/azure_vmss.go:848: Verbose.Infof format %q reads arg #1, but call has 0 args
pkg/cloudprovider/providers/azure/azure_vmss_cache.go:102: Warningf format %q has arg vm.Name of wrong type *string
```

This PR fixes all of those errors, and more in other directories.

The line we're commenting on is causing the final reported error. Leaving the current line unmodified will mean that this test starts failing when Kubernetes updates to Go 1.11. 

I am happy to use any line you suggest provided it does NOT have a printf format error.

Comment (andyzhangx) [reply pkg/cloudprovider/providers/azure/azure_vmss_cache.go]: ok, then let's keep this change, there is not good way since we cannot ensure `vm.Name` is not nil

Comment (rsc) [reply pkg/cloudprovider/providers/azure/azure_vmss_cache.go]: Thanks!

**Comment by idealhack** [comment]
@kubernetes/sig-cluster-lifecycle-pr-reviews

File: pkg/cloudprovider/providers/vsphere/nodemanager.go
Comment (neolit123) [inline pkg/cloudprovider/providers/vsphere/nodemanager.go]: @rsc 
please, change this one to:

```
glog.V(4).Infof("Error while looking for vm=%+v in vc=%s and datacenter=%s: %v",
	vm, res.vc, res.datacenter.Name(), err)
```
/cc @frapposelli 
/cc @kubernetes/sig-cloud-provider-pr-reviews

File: pkg/cloudprovider/providers/azure/azure_loadbalancer.go
Comment (neolit123) [inline pkg/cloudprovider/providers/azure/azure_loadbalancer.go]: seems good in the context of this PR.
if the load balancer name is _really_ wanted here it would need a small refactor.

/cc @justaugustus
/cc @kubernetes/sig-cloud-provider-pr-reviews

File: pkg/cloudprovider/providers/azure/azure_client.go
Comment (neolit123) [inline pkg/cloudprovider/providers/azure/azure_client.go]: these two seems good.
`%v` has to handle a struct that contains a slice of string pointers here.

/cc @justaugustus 
/cc @kubernetes/sig-cloud-provider-pr-reviews

File: pkg/cloudprovider/providers/aws/aws.go
Comment (neolit123) [inline pkg/cloudprovider/providers/aws/aws.go]: `VolumeId` is a `*string`, so this seems good.

/cc @justinsb
/cc @kubernetes/sig-cloud-provider-pr-reviews

**Review by neolit123 — COMMENTED** [review]
thanks @rsc ,

the vsphere change needs a small update.
notified CP leads and sig-cloud-provider, just in case. the rest of the changes seem good.

File: pkg/cloudprovider/providers/vsphere/nodemanager.go
Comment (frapposelli) [reply pkg/cloudprovider/providers/vsphere/nodemanager.go]: 👍 we need to report the `vm` reference to the user for troubleshooting.

**Review by frapposelli — COMMENTED** [review]
Reviewing `vsphere` cloud provider code, please implement @neolit123 change request (https://github.com/kubernetes/kubernetes/pull/65649#discussion_r200224702).

Everything else LGTM

**Review by luxas — APPROVED** [review]
/approve
for cmd/kubeadm

File: pkg/cloudprovider/providers/aws/aws.go
Comment (nckturner) [reply pkg/cloudprovider/providers/aws/aws.go]: Agreed, 
```
type Volume struct {
...
    // The ID of the volume.
    VolumeId *string `locationName:"volumeId" type:"string"`
...
}
```
from https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#Volume

**Review by nckturner — APPROVED** [review]
pkg/cloudprovider/providers/aws looks good.

**Comment by hogepodge** [comment]
sig-cloud-provider will defer to cloud sig leads for approval. Looks like there are changes requested form the vsphere team

**Comment by neolit123** [comment]
@rsc Russ, let me know if you don't have the bandwidth to rebase / amend here. i can only push commits on top of the branch, though, because we don't want to lose this PR.

**Comment by rsc** [comment]
@neolit123 or anyone else, if you can take over updating this PR, please feel free. I checked the "maintainers can modify the PR" box when I created it, or I can add you as a collaborator to my k8s fork. Let me know if you need anything else from me. Thanks.

**Comment by neolit123** [comment]
@rsc i can do that. not sure if the collaborator on fork is needed the last time i've checked, but i'm not sure. you can add me just in case. thanks.

**Comment by rsc** [comment]
@neolit123, done.

**Comment by andrewsykim** [comment]
Thank you @rsc! 

/approve 

for `pkg/cloudprovider`.

**Comment by cblecker** [comment]
/assign
I've rebased this.

**Comment by BenTheElder** [comment]
Pinging some approvers...

Kubelet:
/assign @Random-Liu @yujuhong

IPVS:
/assign @m1093782566

Volume and Mount:
/assign @jsafrane 

PTAL, this PR is exlusively format string fixes and should not take long to review.

**Comment by neolit123** [comment]
amended the requested vsphere change:
https://github.com/kubernetes/kubernetes/pull/65649#discussion_r200224702

re-applying LGTM.
/lgtm

**Comment by frapposelli** [comment]
👍 from the vSphere side

/lgtm

**Comment by m1093782566** [comment]
/lgtm 

from IPVS side.

**Comment by yujuhong** [comment]
/lgtm

for kubelet changes.

**Comment by fejta** [comment]
/assign @jingxu97 @saad-ali @jsafrane 

Please review and `/approve`

**Comment by brendandburns** [comment]
/approve

(and LGTM from the Azure side)

**Comment by neolit123** [comment]
thanks everyone. :+1:

**Comment by rsc** [comment]
Thanks very much @neolit123.

**Pull request merged**
